### ISSUE TITLE
Error: main.js from Terser Name expected (on serve/build)

### ISSUE THREAD

**Issue by onezerohero** [angular/angular-cli#22518]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

Hello, apologies if I'm reporting this in the wrong place - I'm really not sure what's going on.

I am experiencing a Terser error when I try to serve or build my app. 

If I set optimization: false in angular.json, the app will build fine. 

The issue appears to be linked to the AWS-SDK. I have a component that imports the aws sdk and sets config like so:

`AWS.config.update({
      region: "us-east-1"
    });`
    
If this code is commented out, the app will build fine (even with optimization: true, that is). 

### Please provide a link to a minimal reproduction of the bug

https://github.com/peasantzero/repro-app

This project reproduces the error - if one clones the repo and tries to ng serve, they will see the Terser error. Note: in this minimal reproduction, the error displayed is slightly different - 

Error: vendor.js from Terser
Name expected [vendor.js:80173,5]

instead of the one I'm getting in my actual app,

Error: main.js from Terser
Name expected [main.js:144848,5].

However - surely the same or a closely related issue is affecting both projects.

### Please provide the exception or error you saw

```true
Error: main.js from Terser
Name expected [main.js:144848,5]
    at js_error (.../node_modules/terser/dist/bundle.min.js:550:11)
    at croak (.../node_modules/terser/dist/bundle.min.js:1274:9)
    at as_symbol (.../node_modules/terser/dist/bundle.min.js:3119:27)
    at vardefs (.../node_modules/terser/dist/bundle.min.js:2266:29)
    at var_ (.../node_modules/terser/dist/bundle.min.js:2286:27)
    at statement (.../node_modules/terser/dist/bundle.min.js:1513:28)
    at _embed_tokens_wrapper (.../node_modules/terser/dist/bundle.min.js:1339:26)
    at block_ (.../node_modules/terser/dist/bundle.min.js:2172:20)
    at _function_body (.../node_modules/terser/dist/bundle.min.js:2084:21)
    at arrow_function (.../node_modules/terser/dist/bundle.min.js:1692:20)
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 12.1.4
Node: 14.16.1
Package Manager: npm 7.24.0
OS: darwin x64

Angular: 12.1.5
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1201.4
@angular-devkit/build-angular   12.1.4
@angular-devkit/core            12.1.4
@angular-devkit/schematics      12.1.4
@angular/cdk                    12.1.4
@angular/cli                    12.1.4
@angular/material               12.1.4
@schematics/angular             12.1.4
rxjs                            6.6.7
typescript                      4.3.5
```

### Anything else?

I have no idea what's going on here, any and all assistance/direction is much appreciated!

**Comment by alan-agius4** [comment]
Can you setup a minimal repro please?

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

**Comment by onezerohero** [comment]
Minimal repro added!

**Comment by alan-agius4** [comment]
Hi @onezerohero, 

I downloaded the repro but I didn't get any build error when running `ng serve` nor `ng serve --configuration=production`.

**Comment by onezerohero** [comment]
Hi @alan-agius4 - odd. From a coworker's advice, I tried deleting my node_modules and package-lock.json, running npm install again, and this does seem to fix the issue - I'm able to serve/build and not getting the error anymore. 

I did the same thing with the repro-app and no longer got the error there either.

I still have no idea what the actual issue was - but it does seem that I'm able to move past it now.
